### PR TITLE
#1338 Combine algorithm and function selection for creating tasks

### PR DESCRIPTION
**Pull request by rian-schobers**
image

### CODE REVIEW

**Review by bartvanb — COMMENTED** [review]
I have very minor comments on the code, only one textual improvement and one question.

I did have one other finding in testing this: the alignment of the boolean arguments with the text is not the same as before (I checked it with branch `release/4.9` so I guess the issue is on this branch).
image
Before, the checkboxes were below the explanation. Could you fix that?

File: vantage6-ui/src/app/pages/analyze/task/create/task-create.component.ts [hunk 1]
```diff
@@ -245,8 +253,30 @@ export class TaskCreateComponent implements OnInit, OnDestroy, AfterViewInit {
...
 
-  getAlgoSpec(algorithm: Algorithm): string {
-    return `${algorithm.id}_${algorithm.algorithm_store_id}`;
+  search() {
+    const value = this.functionForm.controls.algorithmFunctionSearch.value;
+    this.filteredFunctions = this.functions.filter((func) => {
+      const curAlgorithm = this.algorithms.find((_) => _.id === func.algorithm_id && _.algorithm_store_id == func.algorithm_store_id);
+      const storeName = curAlgorithm ? this.getAlgorithmStoreName(curAlgorithm) : '';
+      return [func.algorithm_name, func.type, storeName, func.display_name, func.name].some((val) =>
+        val?.toLocaleLowerCase()?.includes(value.toLowerCase())
```
[the remark this answers is not in the document]
Comment (rian-schobers) [reply vantage6-ui/src/app/pages/analyze/task/create/task-create.component.ts:262]: Oops, my mistake. I'll change it.

**Comment by rian-schobers** [comment]
Aangepast:
image

File: vantage6-ui/src/app/pages/analyze/task/create/task-create.component.ts [hunk 2]
```diff
@@ -272,7 +272,8 @@ export class TaskCreateComponent implements OnInit, OnDestroy, AfterViewInit {
   getFunctionOptionLabel(func: AlgorithmFunctionExtended): string {
     const curAlgorithm = this.algorithms.find((_) => _.id === func.algorithm_id && _.algorithm_store_id == func.algorithm_store_id);
     const storeName = curAlgorithm ? this.getAlgorithmStoreName(curAlgorithm) : '';
-    return `${func.display_name ?? func.name} <div class="detail-txt"> (${func.algorithm_name}, ${func.type}), ${storeName}</div>`;
+    console.log(func.display_name ?? func.name, func.algorithm_name, func.type, storeName);
```
Comment (bartvanb) [inline vantage6-ui/src/app/pages/analyze/task/create/task-create.component.ts:275]: Can you please get rid of this and then merge?

**Review by bartvanb — APPROVED** [review]

**Pull request merged**